linker/shared/src/main/scala/org/scalajs/linker/backend/WebAssemblyLinkerBackend.scala

```diff
@@ -32,6 +32,10 @@ import org.scalajs.linker.backend.wasmemitter.Emitter
 final class WebAssemblyLinkerBackend(config: LinkerBackendImpl.Config)
     extends LinkerBackendImpl(config) {
 
+  require(
+    coreSpec.esFeatures.esVersion >= ESVersion.ES2015,
+    s"The WebAssembly backend requires at least ES2015; was ${coreSpec.esFeatures.esVersion}."
+  )
```

Comment:
I don't understand how this check relates to this PR.

Reply:
Ah yes ... it's not. I thought it was at some point, but in the end that's a different thing. And it's not even about `esVersion` per se, but rather about `useECMAScript2015Semantics`. I'll make a separate PR later about that. I removed it from this one.